### PR TITLE
Support link underline preference

### PR DESCRIPTION
**Pull request by mperrotti**
This was initiated by our need for underlined links in blocks of text, and supports dotcom's new user setting to render links with underlines. The link underline setting is currently being tested on dotcom behind a feature flag.

A new setting has been added to the Storybook toolbar that allows us to preview what a component would look like when the user has specified that they prefer link underlines.

Closes https://github.com/github/primer/issues/2536

#### User who turned link underlines on in their GitHub settings
Screenshot 2023-09-14 at 5 45 07 PM

### Additional context about link underline settings

Issue: https://github.com/github/releases/issues/3340
Dotcom PR: https://github.com/github/github/pull/278701

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by mperrotti** [comment]
There are cases where we want links in a single like of text to be underlined. Converting this to a draft until I update this PR to address those cases.

Examples:
CleanShot 2023-09-08 at 16 00 14@2x
CleanShot 2023-09-08 at 16 01 01@2x
Screenshot 2023-09-08 at 10 05 19 AM

**Comment by mperrotti** [comment]
Just ignore this PR until I mark it as "Ready for review".

We have more changes to make after talking with @ericwbailey about the feature-flagged user preferences for link underlines.

File: src/drafts/Prose/Prose.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import React from 'react'
```
Comment (mperrotti) [inline src/drafts/Prose/Prose.tsx:1]: I'm questioning whether PRC even needs this component. It's very rare that we need to render a large block of prose in the apps that use Primer.

Reviewers - do you think we should add a Prose component?

Comment (joshblack) [reply src/drafts/Prose/Prose.tsx:1]: @mperrotti I'd be down to ship with the underline changes that you have with the custom property stuff and then see if we still need something like this in the future 👀 Like you said, it seems like underline behavior is more of a one-off thing.

**Comment by joshblack** [comment]
Just wanted to check-in on this @mperrotti to see what you thought of: https://github.com/primer/react/pull/3720#discussion_r1330528288 and how you wanted me to move forward with reviewing this PR 👀

File: src/drafts/Prose/Prose.tsx (hunk 1 above)
Comment (mperrotti) [reply src/drafts/Prose/Prose.tsx:1]: Thanks @joshblack - I totally missed your comment last week. I'll take out the Prose stuff.

File: .playwright/snapshots/components/Breadcrumbs.test.ts-snapshots/Breadcrumbs-Default-light-forcedUnderlines-linux.png
Comment (joshblack) [inline .playwright/snapshots/components/Breadcrumbs.test.ts-snapshots/Breadcrumbs-Default-light-forcedUnderlines-linux.png:1]: @mperrotti just wanted to double-check the snapshot here, it seems like we should be seeing underlines in this snapshot, right?

Comment (mperrotti) [reply .playwright/snapshots/components/Breadcrumbs.test.ts-snapshots/Breadcrumbs-Default-light-forcedUnderlines-linux.png:1]: Yes, we should. I'll investigate...

Comment (mperrotti) [reply .playwright/snapshots/components/Breadcrumbs.test.ts-snapshots/Breadcrumbs-Default-light-forcedUnderlines-linux.png:1]: It looks correct in the deployment preview. Any idea what might be causing the snapshots to be wrong?

Comment (joshblack) [reply .playwright/snapshots/components/Breadcrumbs.test.ts-snapshots/Breadcrumbs-Default-light-forcedUnderlines-linux.png:1]: @mperrotti let me take a look and see 🤔 something might be off on the VRT side.

**Review by joshblack — APPROVED** [review]
I _think_ that last commit should do the trick once the screenshots update 🤞 Looks great!

**Pull request merged**